quarkchain/cluster/shard_state.py

```diff
@@ -806,14 +806,20 @@ def __rewrite_block_index_to(self, minor_block, add_tx_back_to_queue=True):
             self.db.remove_minor_block_index(block)
             if add_tx_back_to_queue:
                 self.__add_transactions_from_block(block)
+        if len(old_chain) > 0:
             asyncio.ensure_future(
-                self.subscription_manager.notify_log(block.header, True)
+                self.subscription_manager.notify_log(block.header, len(old_chain), True)
```

Comment:
`old_chain` seems to be descending (see line 799 and 802), the here `block.header` is the last element of `old_chain` which is the one with lowest height. this seems wrong?